### PR TITLE
GrafanaUI: Adjust misalignment of Checkbox

### LINKED ISSUE #68570: Grafana UI: Misaligned `Checkbox` inside of a `Stack` (experimental) with `flexDirection: column` in Safari
### 
# What went wrong?

**What happened**:

This issue: https://github.com/grafana/grafana/issues/68293 happened, and during its investigation we encountered a possible issue with having a `Checkbox` inside a `Stack` which has its `flexDirection` set to `column`. 
(In our case there's no problem in setting the `flexDirection` to `row`, which makes the `Checkbox` be aligned correctly, so that's the fix we went for.)

https://github.com/grafana/grafana/assets/13227501/3d650dac-2dbe-4b4a-871f-27072648aa46

It seems that the `display: inline-grid` of the `label` of the `Checkbox` (grafana/ui) is not aligning itself well inside a `Stack` (grafana/experimental) component, when the `flexDirection` of its `div` is `column` instead of a `row`. 

**What did you expect to happen**:

  - The Checkbox inside a Stack is aligned correctly, regardless of the Stack's flex direction settings. 

### 
# How do we reproduce it?

After having reverted the partial fix in https://github.com/grafana/grafana/pull/68565 : 
 
**Step 1**:

  - Open any dashboard in Safari

**Step 2**:

  - Change either a variable or time range selection

**Step 3**:

  - Open the sidebar to save the dashboard

**Step 4**:

  - See the misalignment between the label, input field, span element, and the other elements in the form.

image

### 
# What Grafana version are you using?

10.0.0

### 
### Is the bug inside a Dashboard Panel?

Copy the panel's ["get-help" data](https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/) here

### Grafana Platform?

None

### User's OS?

MacOS Ventura 13.2.1

### User's Browser?

Safari 16.3

### Is this a Regression?

None

### Are Datasources involved?

_No response_

### Anything else to add?

_No response_

### PR DESCRIPTION
**Pull request by L-M-K-B**
**What is this feature?**

When being in a stack with other components the Checkbox got aligned wrongly. That's what gets fixed with this PR. We removed all grid related styling and added a container for label and description.

**Why do we need this feature?**

In order to align the Checkbox properly in a stack.

**Who is this feature for?**

Everyone.

**Which issue(s) does this PR fix?**:

Fixes #68570

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.

### CODE REVIEW

**Review by eledobleefe — APPROVED** [review]
I checked it again and seems working. Good mob 🎉

**Review by ashharrison90 — CHANGES_REQUESTED** [review]
this changes the behaviour of checkbox. compare local storybook to developers.grafana.com:

| before | after |
| --- | --- |
| image | image |

notice how the checkbox is center aligned with the title in the first row, whereas afterwards it's center aligned with both the title and description

File: packages/grafana-ui/src/components/Forms/Checkbox.tsx [hunk 1]
```diff
@@ -75,11 +81,15 @@ export const getCheckboxStyles = (theme: GrafanaTheme2, invalid = false) => {
...
     wrapper: css`
-      display: inline-grid;
-      align-items: center;
-      column-gap: ${theme.spacing(labelPadding)};
-      position: relative;
-      vertical-align: middle;
+      display: inline-flex;
+      align-items: flex-start;
+      flex-direction: column;
+      & div:nth-child(1) {
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/Forms/Checkbox.tsx:87]: we should definitely not do this. this applies to any first child of **every** div that's a child of this element 😬 should be able to take a look after lunch

**Review by ashharrison90 — CHANGES_REQUESTED** [review]

File: packages/grafana-ui/src/components/Forms/Checkbox.tsx (hunk 1 above)
Comment (eledobleefe) [reply packages/grafana-ui/src/components/Forms/Checkbox.tsx:87]: 😬 I was not sure about it, that was why I asked you for checking it

**Comment by ashharrison90** [comment]
can we just give the checkbox a default alignment? then the only change needed is to add `align-self: flex-start` to the wrapper class. feels safer? 🤔

**Comment by L-M-K-B** [comment]
I cleaned up the styling even more and it looks good now. @ashharrison90 and @eledobleefe what do think?

**Review by ashharrison90 — CHANGES_REQUESTED** [review]
hate to be the bearer of bad news, but...

1. checkbox shrinks when shrinking the container:
image
could probably fix this by setting `flex-shrink: 0;` on the span. 

1. the `input` itself is now not sized correctly:
image

1. it's also still a change in behaviour (maybe this is fine?) because the checkbox aligns at the start of the text instead of the center:

| before | after |
| --- | --- |
| image | image |

**Comment by L-M-K-B** [comment]
Solved by #70673

**Pull request closed without merging**